### DISCUSSION TITLE
Q: How to find source file by 'datafile'

### DISCUSSION THREAD

**Discussion by struthio** [borgbackup/borg#7921] [Q&A]
hi,
I use borg to backup rather big folder (around 5TB).
I accidentally deleted one file 'data/13/13464', is there a way to find out to which file this chunk belonged, so I can rerun backup for just missing file/files ? I know full backup should automatically fix this, but it takes few days (with lzma compression), and I would like to avoid that as much as possible.

**Comment by ThomasWaldmann** [comment]
You should run `BORG_PASSPHRASE=xxxxx borg check --repair -v --progress ...` on that repo.

The env var for the passphrase will work around the problem that borg will ask you for the passphrase after some hours or days when it starts the archives part of the check.

After that is completed, borg will have removed defective archives and replaced missing file content chunks by all-zero replacement chunks and the repo will be in a consistent state again.

Then run all your usual backup(s). It will only compress and write data to the repo that is not in the repository already. Files that did not change at all should be processed very fast due to borg's files cache.

After that, run the check/repair again. If some missing chunks were reproduced by the new backup(s), it might heal some files that have replacement chunks.

**Comment by struthio** [comment]
Ok so only option is check on whole repo. I will do that. Thanks

**Comment by struthio** [comment]
Is there any way of check how much time left for full backup is left ?
I've done like You said. executed repair - which took rather fast (about 1-2 days), then on 2023-11-13 I've executed full backup again.
It very fast went to '79% Syncing chunks cache. Processing archive FullDrive_2023-11-06T12:52:37' and in this state it is running for a few days already. I see some network activity, and changes in lsof, but progress above not increased for 5 days and I am starting to thinking it hanged somehow.

**Comment by ThomasWaldmann** [comment]
borg displays the progress (see `--progress` option) - that is something it can know for sure because it refers to stuff already done, but it does not do guesswork about the future. see the xkcd about that. :-)

the time the cache sync needs per archive depends in the archive size (amount of files / chunks in the archive).
